DSpace/DSpace
# Feature curation link checker customizations

## Description
Fixes an issue with the link checker in DSpace which fails if a link provided ends up being redirect to a new one and follows the link making sure both resolve correctly.

## Instructions for Reviewers

* TestDspaceRunnableHandler
  * DSpace-api already has this for its Runnable Handler we are just implementing this improvment here which allows for us to view debug logs that the runner encountered 
*  BasicLinkChecker
   * Set the connection instance to follow redirects and then follow that and view its status.

To test take the following steps:

- Make a new dspace item with a metadata field that redirects.
- Check that when running the `check links` curation task the URI redirect link passes

## Checklist
